voidcrew/mapping/docking_port/_docking.port.dm

```diff
@@ -1,13 +1,68 @@
+/**
+ * The main docking port that all voidcrew ships should be using.
+ */
 /obj/docking_port/mobile/voidcrew
-	width = 12
-	dwidth = 5
-	height = 7
+	launch_status = UNLAUNCHED
+
+	/// Makes sure we dont run linking logic more than once
+	VAR_PRIVATE/cached_z_level
+	var/z_levels_above = 0
+	var/z_levels_below = 0
 
 	///The linked overmap object, if there is one
-	var/obj/structure/overmap/ship/current_ship //voidcrew todo: ship functionality, this is never linked!!
+	var/obj/structure/overmap/ship/current_ship
+
 	///List of spawn points on the ship.
 	var/list/obj/machinery/cryopod/spawn_points = list()
 
+/obj/docking_port/mobile/voidcrew/Destroy(force)
+	current_ship.shuttle = null
+	current_ship = null
+	spawn_points.Cut()
+	unlink_from_z_level()
+	return ..()
+
+/obj/docking_port/mobile/voidcrew/connect_to_shuttle(mapload, obj/docking_port/mobile/port, obj/docking_port/stationary/dock)
+	unlink_from_z_level()
+	link_to_z_level()
```

Comment:
Why are we calling both of these lol, it shouldn't have anything to unlink from when connecting to a shuttle, no?